wasm/instructions_test.cpp

```diff
@@ -26,7 +27,8 @@ std::optional<InsnVec> parse(std::string s) {
 // https://webassembly.github.io/spec/core/binary/instructions.html
 int main() {
     // NOLINTBEGIN(modernize-raw-string-literal): This is hex data, not 'A'.
-    etest::Suite s{"wasm::instructions"};
+    etest::Suite s{"wasm::instructions parsing"};
+    etest::Suite t{"wasm::instructions serialization"};
```

Comment:
I think it'd be nicer if we move this to a separate file, or at least a separate function like you mentioned. Having the parsing and serialization tests interspersed like this is a bit hard to read, and a bonus is that your new tests don't need the clang-tidy nolint applied to all the parsing tests.

I think I'd prefer a separate file since they're very different functionalities, but it's up to you.